fix: metadata readme description csv parser

JackSharebourg opened the pull request:
### Required for all PRs:

<!-- Complete the tasks in the following list. Change [ ] to [x] to
show completion. -->

- [x] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

<!-- Link to issues that describe the need for the change. Issues
should include context that will help reviewers understand why the
change is needed.

Make sure to link issues and using a keyword like "resolves #1234".
https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

Metadata is stored in fields rather than tags by default as is already described in the doc. Metadata can still be used as tag via the `csv_tag_columns` option. Original feat #10083 hasn't been released yet. This will be a breaking change after it is in `v1.22` or `v1.21.5`!
Fixed misspelled optionnames and deleted unnecessary linebreak in the readme.

srebhan reviewed (commented):
Thanks @JackSharebourg for looking into this! However, I'm not exactly sure if we should add this complexity here. With the current code, you can convert metadata to fields using the [converter processor](https://github.com/influxdata/telegraf/tree/master/plugins/processors/converter) and I think that putting values into metadata is ... special to be honest. What do you think?

JackSharebourg commented:
Well I agree with you @srebhan that putting values in metadata is special :D. But nevertheless I see two points why metadata should go into fields by default. First of all, this is the standard behavior of the csv parser. Everything is a field unless you specifically configure it otherwise. And second, metadata is IMO the ideal place for the measurement name, but the measurement name can only be chosen from fields. The latter one is for me the decisive factor.
This way you have all the flexibility to use metadata as you please. By using the converter processor all of this is, of course, also possible when metadata is a tag by default. But you have to use an extra processor which adds more complexity to the configuration and is IMO much more error prone. 

After #10219 implemented a prioritization, in one of the special cases, metadata could also be used as a default value. A field is specified in the metadata, but this value can be overwritten with row data if needed.

srebhan commented:
@JackSharebourg, you can use the [converter processor](https://github.com/influxdata/telegraf/tree/master/plugins/processors/converter) to rename the measurement from tag (see the last [example](https://github.com/influxdata/telegraf/tree/master/plugins/processors/converter#example)), can't you? So I don't think this is an argument for your suggestion. :-)

The main reason on why I'm a bit hesitant here is that

1. _Meta-data_ is meta-data and not a metric. We usually keep meta-data in tags, so doing it otherwise in CSV will cause surprise and that's not a good thing.
2. There is the [converter processor](https://github.com/influxdata/telegraf/tree/master/plugins/processors/converter) to convert between tags and fields (and measurement name), so you can satisfy your requirements with the existing setup.
3. When adding meta-data as fields you need to specify types as sooner than later people will ask, "why is this a string, I want this to be an integer". Doing type-mapping requires a more complex configuration and _much more_ complex code and currently we don't have a lack of complexity in this parser.

Looking forward to your rely!

srebhan commented on plugins/parsers/csv/parser.go:
```diff
@@ -351,9 +351,16 @@ outer:
+outerMetadata:
 	for k, v := range p.metadataTags {
-		tags[k] = v
+		for _, tagName := range p.TagColumns {
+			if k == tagName {
+				tags[k] = v
+				continue outerMetadata
+			}
+		}
```
Please use the `internal/choice` package here
```suggestion
		if choice.Contains(k, p.TagColumns) {
			tags[k] = v
			continue
		}
```

srebhan reviewed (commented):
Nice code @JackSharebourg! Please check my comment in case Influx people decide to merge this.

JackSharebourg commented:
Thanks for your feedback @srebhan i hadn't thought about datatypes. This is a good point against my proposed change. Therefore i reverted the code, so metadata remains a tag by default and changed the description in the README to reflect this behavior.

Hipska commented on plugins/parsers/csv/README.md:
```diff
@@ -36,12 +36,11 @@ values.
-  ## Please note that by default, the (key, value) pairs will be added as fields. 
-  ## Use the tag_columns to convert the metadata into tags.
+  ## Please note that by default, the (key, value) pairs will be added as tags. 
+  ## Use the converter processor to use metadata otherwise.
```
I don't know if "otherwise" is the correct word here.

powersj replied:
```suggestion
  ## If fields are required, use the converter processor.
```
